app/src/migration/java/org/mozilla/fenix/MigrationTelemetryListener.kt

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.migration.state.MigrationProgress
+import mozilla.components.support.migration.state.MigrationStore
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+
+class MigrationTelemetryListener(
+    private val metrics: MetricController,
+    private val store: MigrationStore
+) {
+    @UseExperimental(ExperimentalCoroutinesApi::class)
+    fun start() {
+        // Observe for migration completed.
+        store.flowScoped { flow ->
+            flow.collect { state ->
+                Logger("MigrationTelemetryListener").debug("Migration state: ${state.progress}")
+                if (state.progress == MigrationProgress.COMPLETED) {
+                    metrics.track(Event.FennecToFenixMigrated)
+                }
+            }
```

Comment:
This should work. As an alternative we could also emit a `Fact` from the component. With that you wouldn't need to listen here indefinitely.

Reply:
@pocmo if that is easy let's do it. I'm going to land this for now. I filed https://github.com/mozilla-mobile/android-components/issues/5929, if that lands I'll factor out this code and use the Fact instead 👍

Reply:
I've put up a PR for this: https://github.com/mozilla-mobile/android-components/pull/5932